thorlcr/master/thgraphmanager.cpp

```diff
@@ -1001,12 +1002,52 @@ bool CJobManager::executeGraph(IConstWorkUnit &workunit, const char *graphName,
         } cBlock(activeTasks);
 
         {
+#ifdef _CONTAINERIZED
+            // collate pod distribution
+            VStringBuffer selector("thorworker-%s-%s", wuid.get(), graphName);
+            std::vector<std::vector<std::string>> pods = getPodNodes(selector.toLowerCase());
+            std::unordered_map<std::string, unsigned> podPerNodeCounts;
+            for (const auto &podNode: pods)
+            {
+                const std::string &node = podNode[1]; // pod is 1st item, node is 2nd
+                podPerNodeCounts[node]++; // NB: if doesn't exist is created with default value of 0 1st
+            }
+            const StatisticsMapping podStatistics({StNumPods});
+            CRuntimeSummaryStatisticCollection podStats(podStatistics);
+            unsigned nodes=0;
+            double sum=0;
+            double sumSquares=0;
+            for (auto &node: podPerNodeCounts)
+            {
+                podStats.setStatistic(StNumPods, node.second, nodes++);
+                sum += node.second;
+                sumSquares += node.second*node.second;
+            }
+            double mean = sum / nodes;
+            double variance = (sumSquares - sum * mean) / nodes;
+            double stdDev = sqrt(variance);
+            // calculate the above, before locking the workunit below to avoid holding lock whilst issuing getPodNodes call
+#endif
+
             Owned<IWorkUnit> wu = &workunit.lock();
             wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTgraph, graphScope, StWhenStarted, NULL, startTs, 1, 0, StatsMergeAppend);
             //Could use addTimeStamp(wu, SSTgraph, graphName, StWhenStarted, wfid) if start time could be this point
             wu->setState(WUStateRunning);
             VStringBuffer version("%d.%d", THOR_VERSION_MAJOR, THOR_VERSION_MINOR);
             wu->setDebugValue("ThorVersion", version.str(), true);
+
+#ifdef _CONTAINERIZED
+            // issue warning and publish pod distribution stats, if any stddev
+            if (stdDev)
+            {
+                Owned<IStatisticGatherer> collector = createGlobalStatisticGatherer(wu);
+                StatsScopeId graphScopeId(graphName);
+                collector->beginScope(graphScopeId);
+                podStats.recordStatistics(*collector, false);
+                Owned<IException> e = makeStringExceptionV(-1, "Degraded performance. Worker pods are unevenly distributed over nodes. StdDev=%.2f. See workunit stats. for more info.", stdDev);
```

Comment:
This doesn't tell you which slave pods were on the nodes that had fewer or more pods.  Is that going to be something you wish you knew?